src/hooks/screen/useMessageHook.ts

```diff
@@ -0,0 +1,86 @@
+import {Alert, NativeSyntheticEvent} from 'react-native';
+import {ContextMenuOnPressNativeEvent} from 'react-native-context-menu-view';
+import {useCallback, useContext} from 'react';
+
+import AnonymousMessageRepo from '../../service/repo/anonymousMessageRepo';
+import ChatSchema from '../../database/schema/ChatSchema';
+import ShareUtils from '../../utils/share';
+import SignedMessageRepo from '../../service/repo/signedMessageRepo';
+import UseMessageHook from '../../../types/hooks/screens/useMessageHook.types';
+import useChatScreenHook from './useChatScreenHook';
+import useLocalDatabaseHook from '../../database/hooks/useLocalDatabaseHook';
+import {ANONYMOUS} from '../core/constant';
+import {Context} from '../../context';
+import {calculateTime} from '../../utils/time';
+import {setReplyTarget} from '../../context/actions/chat';
+
+function useMessageHook(): UseMessageHook {
+  const {localDb, refresh} = useLocalDatabaseHook();
+  const {handleUserName} = useChatScreenHook(ANONYMOUS);
+  const [replyPreview, dispatch] = (useContext(Context) as unknown as any).chat;
+
+  const setReplyPreview = useCallback((messageItem) => {
+    console.log('Reply Message');
+    setReplyTarget(messageItem, dispatch);
+  }, []);
+
+  const clearReplyPreview = useCallback(() => {
+    console.log('clear Reply target');
+    setReplyTarget(null, dispatch);
+  }, []);
+
+  const deleteMessage = async (messageId: string, type: 'ANONYMOUS' | 'SIGNED') => {
+    if (!localDb) return;
+    try {
+      if (type === 'ANONYMOUS') {
+        await AnonymousMessageRepo.deleteMessage(messageId);
+      } else {
+        await SignedMessageRepo.deleteMessage(messageId);
+      }
+      ChatSchema.updateDeletedChatType(localDb, messageId);
+      refresh('chat');
+    } catch (error) {
+      console.log('error on delete message:', error);
+    }
+  };
+
+  const onContextMenuPressed = (
+    e: NativeSyntheticEvent<ContextMenuOnPressNativeEvent>,
+    data,
+    type: 'ANONYMOUS' | 'SIGNED'
+  ) => {
+    const event = e.nativeEvent;
+    data = data?.message?.user ? data?.message : data;
+
+    if (event.index === 0) {
+      const messageItem = {
+        username: handleUserName(data),
+        time: calculateTime(data?.updated_at, true),
+        message: data?.message ?? data?.text,
+        messageId: data?.id,
+        chatType: type,
+        messageType: 'regular'
+      };
+      setReplyPreview(messageItem);
+    }
+    if (event.index === 1) {
+      ShareUtils.copyTextToClipboard(data?.message);
+    }
+    if (event.index === 2) {
+      Alert.alert('Delete Message?', '', [
+        {text: 'Cancel', style: 'cancel'},
+        {text: 'Confirm', style: 'destructive', onPress: () => deleteMessage(data?.id, type)}
+      ]);
+    }
+  };
+
+  return {
+    replyPreview: replyPreview?.replyTarget,
+    setReplyPreview,
+    clearReplyPreview,
+    onContextMenuPressed
+  };
+}
+
+export default useMessageHook;
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The use of `console.log` in production code is generally discouraged. It can expose potentially sensitive information and impact performance. Consider using a logging library that can be configured to log only in development environments or certain log levels in production.

```diff
  const setReplyPreview = useCallback((messageItem) => {
-24:     console.log('Reply Message');
    setReplyTarget(messageItem, dispatch);
  }, []);

  const clearReplyPreview = useCallback(() => {
-29:     console.log('clear Reply target');
    setReplyTarget(null, dispatch);
  }, []);
```

Also, the error handling in the `deleteMessage` function could be improved. Currently, it just logs the error but doesn't provide any feedback to the user if something goes wrong. Consider adding an alert or some other form of notification to inform the user about the error.

```diff
    } catch (error) {
-44:       console.log('error on delete message:', error);
+44:       Alert.alert('Error', 'There was an error deleting the message.');
    }
```

In line 21, you are casting the result of `useContext(Context)` to `any`. This defeats the purpose of TypeScript's static typing. You should create a type or interface that matches the structure of your context and use that instead.

```diff
-21:   const [replyPreview, dispatch] = (useContext(Context) as unknown as any).chat;
+21:   const [replyPreview, dispatch] = (useContext(Context) as unknown as MyDefinedContextType).chat;
```
Please replace `MyDefinedContextType` with the actual type of your context.

Lastly, in the `onContextMenuPressed` function, you're directly mutating the `data` parameter on line 54. It's generally a bad practice to mutate function arguments as it can lead to unexpected behavior. Consider creating a new variable instead.

```diff
-54:     data = data?.message?.user ? data?.message : data;
+54:     const processedData = data?.message?.user ? data?.message : data;
```
Remember to replace all subsequent references of `data` with `processedData` in the `onContextMenuPressed` function.


<!-- This is an auto-generated comment by OSS CodeRabbit -->